pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeFieldSpec.java

```diff
@@ -70,17 +72,8 @@ public DateTimeFieldSpec() {
    */
   public DateTimeFieldSpec(String name, DataType dataType, String format, String granularity) {
     super(name, dataType, true);
-    Preconditions.checkNotNull(name);
-    Preconditions.checkNotNull(dataType);
-    if (Character.isDigit(format.charAt(0))) {
-      DateTimeFormatSpec.validateFormat(format);
-    } else {
-      DateTimeFormatSpec.validatePipeFormat(format);
-    }
-    DateTimeGranularitySpec.validateGranularity(granularity);
-
-    _format = format;
-    _granularity = granularity;
+    setFormat(format);
```

Comment:
It is covered in `SchemaUtilsTest`.
After a second thought, decided to not directly parse the `format` and `granularity` in the setter to prevent the extra overhead of reading schema. The parsing is performed lazily in the getter.